consensus/README.md

```diff
@@ -140,6 +150,244 @@ graph TB
+
+#### Download routine
+
+* The `download` routine is alive as long as `sync` mode is on
+* It checks in its persistence for the latest downloaded block and tries to get and add to its persistence all the blocks; up to the network current height
+* After downloading, and before inserting the block, basic (stateless) verification is made to the block
+* A downloaded and inserted block is a structurally valid block but should by no mean considered valid w.r.t. its validators signatures or transactions within
+
+```mermaid
+sequenceDiagram
+  title: Consensus: Download blocks
+  participant Persistence
+  Note over Node: (Download routine)
+  participant Node
+  participant Network
+
+  par
+    loop
+      Note left of Network: Constantly asking for the highest<br /> known block from network
+      Node->>Network:UpdateNetworkCurrentHeight
+      Network-->>Node: 
+    end
+  and
+    Node->>Persistence:GetLastDownloadedBlock
+    Persistence-->>Node: 
+
+    loop LastDownloadedBlock < NetworkCurrentHeight
+      Node->>Network:GetBlock
+      Network-->>Node: 
+      Note left of Network: Currently the node asks all the network<br />for the block it wants to download
+
+      Node->>Persistence:Append
+      Persistence-->>Node: 
+      Note left of Node: append if new block
+
+      Node->>Node:Wait(until new block)
+      Note right of Node: If latest network block is reached<br />Wait for a new block to continue
+    end
+  end
+```
+
+### Apply routine
+
+_Note: We do not detail how individual transactions are applied or how state is derived from them. Just assume that the state (specifically validator set) may be mutated after each block application._
+
+* The `apply` routine remains alive as long as `sync` mode is on
+* It needs a starting state (genesis, or loaded local state) to build blocks from
+* Each block is validated and applied before moving to the next one
+* The block application is a sequential process starting from genesis, applying blocks up till network current height
+* Since basic validation is done at download step and assume that the Pocket node trusts its persistence layer, it is safe to skip basic revalidation
+* The `apply` mechanism needs to maintain a chain of trust while applying blocks by performing the following:
+  * Before applying a block `n`, verify that is is signed by a quorum of `n-1`'s validator set (genesis validator set for block `1`)
+  * By applying each block, the validator set is updated (validators joining or leaving), starting from genesis validator set for any new node
+* With this chain of trust, a synching node systematically detects invalid blocks
+  * No malicious or faulty node could inject an alternative block without making at least `2t+1` validators sign it
+  * The persistence layer is mainly a cache for the block application, so a node won't restart block application from genesis each time it's rebooted
```

Comment:
Would be great to know more about others roles persistence have beside "snapshotting" previous derived states.

Reply:
@red-0ne Below are some tips.
cc @dylanlott (owner of the persistence module) 

Request: As you read through this documentation, PLEASE make notes of questions/problems and DO NOT HESITATE to submit a PR to edit it.

---

Below are some tips.

- [ ] 1. Have you read through the persistent documentation in the Wiki? https://github.com/pokt-network/pocket/wiki/Persistence-State-Hash-Protocol

![Screenshot 2023-08-04 at 10 44 44 AM](https://github.com/pokt-network/pocket/assets/1892194/25382fcb-cce1-4a23-9355-d41f502829a4)

- [ ] 2. Have you read through the persistence specification? https://github.com/pokt-network/pocket-network-protocol/
![Screenshot 2023-08-04 at 10 46 12 AM](https://github.com/pokt-network/pocket/assets/1892194/22faa42e-a9ba-4e51-aaa8-25c1fb538061)

- [ ] 3. Have you tried asking gPOKT? https://gpoktn.streamlit.app/

![Screenshot 2023-08-04 at 10 47 34 AM](https://github.com/pokt-network/pocket/assets/1892194/209990bd-d96e-481a-b4d1-b72369f5f7a3)

Reply:
I missed reading the Persistence doc, will dive into it soon.

I see how coining it as "mainly a cache" may be misleading and not fully reflect its role. Also using the term "cache" was meant to put light on the idea that it handles computed data persistence that could be recomputed with block application, starting from genesis.

Maybe since this part was about the Persistence module seen from the POV of the Consensus mode, the term (restoration/restart/check) point would be more appropriate. Any better term is welcome.